libmscore/score.h

```diff
@@ -1340,6 +1346,9 @@ class MasterScore : public Score {
       bool showOmr() const                     { return _showOmr; }
       void setShowOmr(bool v)                  { _showOmr = v;    }
 
+      std::shared_ptr<Avs::AvsOmr> avsOmr() const      { return _avsOmr; }
+      void setAvsOmr(std::shared_ptr<Avs::AvsOmr> omr) { _avsOmr = omr; }
```

Comment:
Align the right braces

Reply:
the rule is not clear, the length of the variables can be different, some short, some long ...  
I look and now this rule is not fulfilled ..

![image](https://user-images.githubusercontent.com/3818029/76141713-9ecb9580-606f-11ea-9173-b10345db57d1.png)

![image](https://user-images.githubusercontent.com/3818029/76141717-af7c0b80-606f-11ea-84cb-2a64a8b4fe13.png)

![image](https://user-images.githubusercontent.com/3818029/76141726-c15dae80-606f-11ea-8a34-5f03eb9a1a38.png)

Reply:
I fixed, but I think this is not a useful rule, it is difficult to keep it, therefore it does not add to the clear, and in some cases, on the contrary, complicates the reading.

Reply:
It's not a _rule_, it's a _convention_. Just because there're other cases which don't follow the convention doesn't mean it's completely meaningless to follow it. Why can it "complicate the reading" anyway?

Reply:
> Why can it "complicate the reading" anyway?   

the first screenshot shows that there may be a mishmash, it’s hard to understand what's there

Reply:
>  Just because there're other cases which don't follow the convention doesn't mean it's completely meaningless to follow   

This may mean that the convention is not true ;)

Reply:
This is not "cargo cult" if everyone sticks to the convention (doesn't un-align vertically adjacent braces, especially ones belonging to a getter and a setter of the same variable). And IMHO this isn't hard.

The point is, if a convention is obviously already established, deliberately breaking it because you personally think it isn't helpful, isn't helpful as well. The more you break it, the code becomes harder to read.

Reply:
> deliberately breaking it because you personally think it isn't helpful

not quite so, I do conclusions seeing what it leads to, including in MU code

Reply:
A small example.  Suppose I named a variable, and after automatically renamed it to the best name - in the end, all these brackets will move and there will be a mishmash

Reply:
When you see a mishmash, fix it -- that's how you help, not ignoring it and add some more mishmash.

Reply:
There is value in consitency, but there is also value in efficiency. It makes sense to align left braces, but insisting that right braces must be aligned is going a step too in my opinion. The cost to developer time in maintaining right-alignment means it is not worth it, not to mention your time @Howard-C in pointing out the issue.

Reply:
How is having to add a few spaces a hassle?

Reply:
Besides, the function names in the left column are all different lengths and there's nothing anybody can do about that.

Reply:
@Jojo-Schmitz
> How is having to add a few spaces a hassle?

I agree that in this case it is not and I personally would have added spaces to match the surrounding functions. But I don't think that the issue is serious enough that I would have called out another developer for not doing it.

If my function was the odd one out and it meant changing all the other functions then I certainly wouldn't bother as it would make the diff needlessly complicated.

Reply:
Sorry, I didn’t want to criticize the code, I just want to talk about my experience.
I already wrote that I do not contain the presence of implementations in the .h files, except for a few very simple exceptions, this causes a number of problems, and it turns out there is one more :)

I repeat, what I think about the existence of an implementation in .h files:

* It is difficult to find a method implementation. The IDE usually has a list of methods in the .cpp file (or by search), and when you do not find the expected method in it, it will confuse you.
* This makes the code inconsistent, part of the implementations in the .cpp, part in the .h, and you do not know where, in fact, you can only guess. When in .h only the declaration, and in the .cpp only the implementation, then everything is clear.
* It's hard to read a .h file, it gets complicated and littered (ove.h nice to read, score.h - very badly readable, it is very large, but implementation also complicates)
* It is difficult to add comments to methods in .h, you cannot write a short comment on the same line (like this `QByteArray readMxl (); // compressed MuzicXml`)
* It is difficult to add verification of input arguments, boundary conditions, state - this requires more than one line, and you have to either not do this, or even litter .h file, or write unreadable code.
* Difficult to debug:
* It is difficult to add logging to see what values and when are assigned
* It is difficult to add debugging code - checking for a specific value, checking for changes in the value. With this, I tormented myself a lot and for a long time when I found out why the elements are not placed where they should, who, when and which sets the positions for the elements.
    
(later, I will do proposals for changing this approach)

Reply:
But there's one main reason for putting one-line functions in .h file, as I've also said many times: putting all functions in .cpp file makes finding multi-line functions very difficult. The .cpp file is very ugly and most of the things in it are repetitious (get, set, get, set...), so it is uncomfortable to browse through. I don't know about you, and I don't have other meanings, but I, for one, think that this reason prevails over all reasons you mentioned above. It's true that your reasons are considerable too, but I don't think any of them causes as much trouble as the one I provided.

Reply:
And again, until your proposal is approved, I think it isn't the right move to deliberately break the established convention just because you personally think it isn't good. Because if your proposal is denied, another developer will have to clean up the "mess" you introduced.

Reply:
> But there's one main reason for putting one-line functions in .h file, as I've also said many times: putting all functions in .cpp file makes finding multi-line functions difficult. You have to use the search tool(s) in your IDE much more often. The .cpp file is very ugly and most of the things in it are repetitious (get, set, get, set...), so it is uncomfortable to browse through.  

I'm a little surprised if the problem is that .cpp getting large, then why are there so many very large classes in the project? In which, in any case, it is difficult to find anything. And there is no struggle with this particular problem?

For myself, I solve this problem as follows: the more significant the method, the higher it is in the .cpp file, the less significant the method, the lower it is, therefore, these set/get at the end of the file do not interfere.

> And again, until your proposal is approved, I think it isn't the right move to deliberately break the established convention just because you personally think it isn't good. Because if your proposal is denied, another developer will have to clean up the "mess" you introduced.   
  
I agree with you, you are right.
The only thing is that there are many unimportant things that can be left to the author to decide. The main thing is that there should be consistent practices in one class / subsystem. That is, I had to make the right align of the bracket, I was mistaken that I did not, but I did not do it simply because I did not notice that the right bracket should be aligned :)